common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/logging/DockerToActivationLogStore.scala

```diff
@@ -76,14 +77,21 @@ class DockerToActivationLogStore(system: ActorSystem) extends LogStore {
                            container: Container,
                            action: ExecutableWhiskAction): Future[ActivationLogs] = {
 
+    // wait for a sentinel only if no container (developer) error occurred to avoid
+    // that log collection continues if the action code still logs after timeout
+    val sentinel = action.exec.sentinelledLogs && !activation.response.isContainerError
+
     container
-      .logs(action.limits.logs.asMegaBytes, action.exec.sentinelledLogs)(transid)
+      .logs(action.limits.logs.asMegaBytes, sentinel)(transid)
       .via(DockerToActivationLogStore.toFormattedString)
```

Comment:
If we change the `DockerToActivationFileLogStore` such that a developer error injects a message that leads to a log collection error, then this log store should have the same behaviour. In that case, we have to inject / append `Messages.logFailure` at this location - or at least before we check for error messages.